### DISCUSSION TITLE
how to get what field is in focus react hook form

### DISCUSSION THREAD

**Discussion by bryanprimus** [react-hook-form/react-hook-form#8913] [Q&A]
I have this component

```ts
 useForm<{companyLogo: string}>()
```

```ts
<label>
  <input
    accept="image/png"
    type="file"
    hidden
    {...restCompanyLogo}
    ref={e => {
      refCompanyLogo(e);
      inputFileRef.current = e;
    }}
  />
  <Button
    variant="outlined"
    color="secondary"
    size="small"
    sx={{ width: "fit-content" }}
    component="span"
    onKeyDown={evt =>
      evt.keyCode === 32 &&
      inputFileRef.current?.click()
    }
  >
    {tCommon["upload-placeholder"](
      tGlossary["company-logo"]
    )}
  </Button>
</label>
```

The field `companyLogo` is required

When I press submit button, the focus will be in companyLogoField `<Input type="file" hidden/>`. My question is, Is there a way to know that the focus currently in companyLogoField?

Because I want to set the focus to the button instead if the focus is now in {...register("companyLogo")}

I try to check `getFieldState("companyLogo")` but there's no indicator in there that says the companyLogo is focused

**Comment by leapful** [comment]
@bryantobing12 : You can use https://developer.mozilla.org/en-US/docs/Web/API/Document/activeElement.

**Reply by bryanprimus** [reply]
so is that mean that we don't have the ability to do that in `react-hook-form`?

**Reply by leapful** [reply]
@bryantobing12 : Yes, react-hook-form does not need to support it because native Web API in browsers already supported that use case. Why do you need to find focus by using react-hook-form if it's easier to check focus by "activeElement".
